backend/functions/lib/use-cases/acms-orders/acms-orders.test.ts

```diff
@@ -0,0 +1,558 @@
+          caseId: childCases[0].caseId,
+          consolidationDate: '2024-01-01',
+          consolidationType: 'substantive',
+        },
+        {
+          caseId: childCases[1].caseId,
+          consolidationDate: '2024-01-01',
+          consolidationType: 'administrative',
+        },
+      ],
+    };
+
+    const caseSummaryMap = new Map<string, CaseSummary>([
+      [leadCase.caseId, leadCase],
+      [childCases[0].caseId, childCases[0]],
+      [childCases[1].caseId, childCases[1]],
+    ]);
+
+    const expectedFromLinks = details.childCases.map((bCase) => {
+      const caseId = leadCase.caseId;
+      const orderDate = bCase.consolidationDate;
+      const consolidationType = bCase.consolidationType as ConsolidationType;
+      return MockData.getConsolidationFrom({
+        override: {
+          caseId,
+          consolidationType,
+          orderDate,
+          otherCase: caseSummaryMap.get(bCase.caseId),
+        },
+      });
+    });
+
+    const expectedHistory: CaseConsolidationHistory[] = [];
+    const allCaseIds = Array.from(caseSummaryMap.keys());
+    allCaseIds.forEach((caseId) => {
+      expectedHistory.push({
+        documentType: 'AUDIT_CONSOLIDATION',
+        before: null,
+        after: {
+          status: 'approved',
+          leadCase,
+          childCases,
+        },
+        updatedBy: ACMS_SYSTEM_USER_REFERENCE,
+        updatedOn: '2024-01-01',
+        caseId,
+      });
+    });
+
+    const expectedToLinks = details.childCases.map((bCase) => {
+      const caseId = bCase.caseId;
+      const orderDate = bCase.consolidationDate;
+      const consolidationType = bCase.consolidationType as ConsolidationType;
+      return MockData.getConsolidationTo({
+        override: { caseId, consolidationType, orderDate, otherCase: leadCase },
+      });
+    });
+
+    jest.spyOn(AcmsGatewayImpl.prototype, 'getConsolidationDetails').mockResolvedValue(details);
+    jest
+      .spyOn(CasesDxtrGateway.prototype, 'getCaseSummary')
+      .mockImplementation((_context, caseId) => {
+        return Promise.resolve(caseSummaryMap.get(caseId));
+      });
+    const useCase = new AcmsOrders();
+    await useCase.migrateConsolidation(context, leadCase.caseId);
+
+    expect(getConsolidationSpy).toHaveBeenCalled();
+    expect(createConsolidationToSpy).toHaveBeenCalledTimes(childCases.length);
+    expect(createConsolidationFromSpy).toHaveBeenCalledTimes(childCases.length);
+    expect(createCaseHistorySpy).toHaveBeenCalledTimes(expectedHistory.length);
+
+    expectedFromLinks.forEach((fromLink) => {
+      expect(createConsolidationFromSpy).toHaveBeenCalledWith(fromLink);
+    });
+    expectedToLinks.forEach((toLink) => {
+      expect(createConsolidationToSpy).toHaveBeenCalledWith(toLink);
+    });
+    expectedHistory.forEach((history) => {
+      expect(createCaseHistorySpy).toHaveBeenCalledWith(history);
+    });
+  });
+
+  test('should histories for consolidations over a date range', async () => {
```

Comment:
'should histories for consolidations over a date range' doesn't make sense to me.  Did we mean "should get (or retrieve) histories for consolidations over a date range"?